src/components/tree/tree.common.ts

```diff
@@ -0,0 +1,22 @@
+import IgcTreeItemComponent from './tree-item';
+
+export const IgcTreeSelectionType = {
+  None: 'none',
+  Multiple: 'multiple',
+  Cascade: 'cascade',
+} as const;
+export type IgcTreeSelectionType =
+  typeof IgcTreeSelectionType[keyof typeof IgcTreeSelectionType];
+
+export interface IgcTreeEventMap {
+  igcSelection: CustomEvent<any>;
+  igcItemExpanding: CustomEvent<any>;
+  igcItemExpanded: CustomEvent<any>;
+  igcItemCollapsing: CustomEvent<any>;
+  igcItemCollapsed: CustomEvent<any>;
+  igcActiveItem: CustomEvent<any>;
```

Comment:
Let's type the events. `void` where the no detail data is passed and the appropriate `<T>` where necessary